front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -11,16 +17,146 @@ import { useMemo } from "react";
 
 import DataSourceResourceSelectorTree from "@app/components/DataSourceResourceSelectorTree";
 import { useParentResourcesById } from "@app/hooks/useParentResourcesById";
+import { orderDatasourceViewByImportance } from "@app/lib/assistant";
 import {
   CONNECTOR_CONFIGURATIONS,
   getConnectorProviderLogoWithFallback,
 } from "@app/lib/connector_providers";
-import { getDisplayNameForDataSource, isFolder } from "@app/lib/data_sources";
+import {
+  getDisplayNameForDataSource,
+  isFolder,
+  isManaged,
+  isWebsite,
+} from "@app/lib/data_sources";
+
+const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
+const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+
+type DataSourceViewsSelectorProps = {
+  owner: WorkspaceType;
+  dataSourceViews: DataSourceViewType[];
+  selectionConfigurations: DataSourceViewSelectionConfigurations;
+  setSelectionConfigurations: Dispatch<
```

Comment:
This type isn't does not look right since we don't pass a state function 🤔

Reply:
Okay I see, not a big fan of exposing this type of logic outside of the component that own the state.

Reply:
Let's live it like this in this PR but we should avoid this complex state management 🙃.